### PR TITLE
Reorganize the book into 3 parts

### PR DESCRIPTION
**Pull request by mark-i-m**
The guide is quite long and contains _tons_ of useful info, but it at the moment, it lacks flow IMHO.

I propose we organize the guide as two parts: (1) Useful info for contributing, and (2) How the compiler works.

(1) is where we would move general knowledge chapters like "about the compiler team", "building", "debugging", "stabilization process", etc. This section would serve as a reference for stuff that is useful regardless of what you are working on.

(2) is where everything else would live. This section is about how the compiler works.

r? @nikomatsakis

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+- [Walkthrough: a typical contribution](./walkthrough.md)
+- [Stabilizing Features](./stabilization_guide.md)
 - [Debugging the Compiler](./compiler-debugging.md)
 - [Profiling the compiler](./profiling.md)
     - [with the linux perf tool](./profiling/with_perf.md)
+- [Coding conventions](./conventions.md)
+
+---
+
+- [Part 2 Intro](./part-2-intro.md)
```
Comment (nikomatsakis) [inline src/SUMMARY.md]: I think we should name this something like "Part 2: How the compiler works", so you can see at a glance what it's about

File: src/SUMMARY.md [hunk 2]
```diff
@@ -1,21 +1,28 @@
 # Summary
 
-- [About this guide](./about-this-guide.md)
+[About this guide](./about-this-guide.md)
+
+---
+
+- [Part 1 Intro](./part-1-intro.md)
```
Comment (nikomatsakis) [inline src/SUMMARY.md]: I think we should name this something like

Part 1: Building, debugging, and contributing to Rustc

although that seems a bit long to me =)

File: src/SUMMARY.md (hunk 1 above)
Comment (mark-i-m) [reply src/SUMMARY.md]: Good idea :+1: 

Done.

File: src/SUMMARY.md (hunk 2 above)
Comment (mark-i-m) [reply src/SUMMARY.md]: Same.

**Comment by nikomatsakis** [comment]
maybe a pithier title for part 1 might just be "contribution guide"

**Comment by mark-i-m** [comment]
I feel like "contribution guide" is to general (isn't the whole thing a contribution guide?). My original title was "General knowledge", which also felt off somehow... Perhaps "General Guidelines" is right?

**Comment by nikomatsakis** [comment]
maybe keep it as is -- the only problem would be if it looks weird when rendered

**Comment by mark-i-m** [comment]
It looks like a network timeout. I will restart the build.

> maybe keep it as is -- the only problem would be if it looks weird when rendered

Ok. I believe the title should just wrap properly. We already have some long titles like "How to build the compiler and run what you built"...

**Pull request merged**